### ISSUE TITLE
[Bug Report][3.0.4] z-index is incorrect (useStack issue)

### ISSUE THREAD

**Issue by Kasheftin** [vuetifyjs/vuetify#16251]
### Environment
**Vuetify Version:** 3.0.4
**Vue Version:** 3.2.45
**Browsers:** Chrome 105.0.0.0
**OS:** Linux x86_64

### Steps to reproduce
- Click on `Show Overlay` button
- Overlay is shown at the very top (z-index 2000), above the v-app-bar (z-index 1004)
- Close the overlay
- Click on `Show Menu` button
- Click on `Show Overlay` button while menu is visible
- Overlay is shown under the v-app-bar because now it has z-index 30

Additionally: v-overaly has optional z-index prop which does not work correctly when any other overlay is opened. Even if set to z-index=50000, 

### Expected Behavior
It should not matter from where the overlay is opened - it should have predictable manageable z-index

### Actual Behavior
z-index affects only the state when there're no overlays opened. Otherwise it will be last overlay zIndex + 10.

### Reproduction Link
[https://codepen.io/kasheftin/pen/MWXRgKx?editors=101](https://codepen.io/kasheftin/pen/MWXRgKx?editors=101)

### Other comments
On my opinion the issue is on this line, https://github.com/vuetifyjs/vuetify/blob/bd36b460a2be90254f53ce64cdedcafb063d9000/packages/vuetify/src/composables/stack.ts#L29

As soon as lastZIndex exists, the external zIndex prop is dropped. 
I think it should be `_zIndex.value = lastZIndex ? Math.max(lastZIndex + 10, +zIndex.value) : +zIndex.value`

**Comment by zd87** [comment]
Yes! I faced a simila issue: I have a form in a dialog. The form contains autocompletes, but autocomplete menus are not visible due to their index (2000) being inferior to the dialog's (2400). BTW, comboboxes don't have this issue, combobox menu IS visible, it's z-index is 2410. 
CSS won't help in this case because z-indexes are there as inline styles and take priority.

**Comment by canbax** [comment]
I also face an issue due to this increase of zIndex but the codepen link isn't showing a reproducible example 

I see no vuetify component rendered in codepen UI and when I check the console I see below error.

> Uncaught ReferenceError: Vue is not defined 
>  at https://cdpn.io/cpe/boomboom/pen.js?key=pen.js-f418b91c-2f66-05a6-c1d4-db61bfa57d9c:1
